tests/integration/metrics/column/test_column_values_match_regex_values.py

```diff
@@ -0,0 +1,69 @@
+import pandas as pd
+
+from great_expectations.datasource.fluent.interfaces import Batch
+from great_expectations.metrics.column.column_values_match_regex_values import (
+    ColumnValuesMatchRegexValues,
+    ColumnValuesMatchRegexValuesResult,
+)
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.metrics.conftest import SQL_DATA_SOURCES, SnowflakeDatasourceTestConfig
+
+COLUMN_NAME = "whatevs"
+BIG_NUMBER = 101
+MATCH_ALL_REGEX = ".+"
+
+DATA_FRAME = pd.DataFrame({COLUMN_NAME: ["abc", "def", "ghi", "1ab2", None]})
+DATA_FRAME_WITH_LOTS_OF_VALUES = pd.DataFrame({COLUMN_NAME: ["A"] * BIG_NUMBER})
+
+SQL_DATA_SOURCES_EXCEPT_SNOWFLAKE = [
+    datasource
+    for datasource in SQL_DATA_SOURCES
+    if not isinstance(datasource, SnowflakeDatasourceTestConfig)
+]
+
+
+class TestColumnValuesMatchRegexValues:
+    @parameterize_batch_for_data_sources(
+        data_source_configs=SQL_DATA_SOURCES_EXCEPT_SNOWFLAKE,
```

Comment:
Do regexes work differently for Snowflake?

Reply:
Yes! https://docs.snowflake.com/en/sql-reference/functions/rlike#usage-notes

Reply:
Specifically, snowflake automatically adds the regex anchor characters, so `ab` -> `^ab$`